include/StringTools.h

```diff
@@ -0,0 +1,294 @@
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    The Mezzanine Engine is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with The Mezzanine Engine.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/* The original authors have included a copy of the license specified above in the
+   'Docs' folder. See 'gpl.txt'
+*/
+/* We welcome the use of the Mezzanine engine to anyone, including companies who wish to
+   Build professional software and charge for their product.
+
+   However there are some practical restrictions, so if your project involves
+   any of the following you should contact us and we will try to work something
+   out:
+    - DRM or Copy Protection of any kind(except Copyrights)
+    - Software Patents You Do Not Wish to Freely License
+    - Any Kind of Linking to Non-GPL licensed Works
+    - Are Currently In Violation of Another Copyright Holder's GPL License
+    - If You want to change our code and not add a few hundred MB of stuff to
+        your distribution
+
+   These and other limitations could cause serious legal problems if you ignore
+   them, so it is best to simply contact us or the Free Software Foundation, if
+   you have any questions.
+
+   Joseph Toppi - toppij@gmail.com
+   John Blackwood - makoenergy02@gmail.com
+*/
+#ifndef MEZZ_Foundation_StringTools_h
+#define MEZZ_Foundation_StringTools_h
+
+#ifndef SWIG
+    #include "DataTypes.h"
+#endif
+
+/// @file
+/// @brief This file includes a collection of common String manipulation utilities.
+
+namespace Mezzanine {
+namespace StringTools {
+    /// @brief Convenience type used by functions in this namespace for a String iterator.
+    using StrIter = String::iterator;
+    /// @brief Convenience type used by functions in this namespace for a const String iterator.
+    using ConstStrIter = String::const_iterator;
+
+    ///////////////////////////////////////////////////////////////////////////////
+    // Character Checks
+
+    /// @brief Checks if a character is a space.
+    /// @param ToCheck The character to be checked.
+    /// @return Returns true if the character is a space, false otherwise.
+    Boole MEZZ_LIB IsSpace(const Char8 ToCheck);
+    /// @brief Checks if a character is a tab.
+    /// @param ToCheck The character to be checked.
+    /// @return Returns true if the character is a tab, false otherwise.
+    Boole MEZZ_LIB IsTab(const Char8 ToCheck);
+    /// @brief Checks if a character is a newline.
+    /// @param ToCheck The character to be checked.
+    /// @return Returns true if the character is a newline, false otherwise.
```

Comment:
What does this do with a \r? It is part of a newline on some platforms.

Reply:
Looking at the source, the function checks for both \r and \n.

Reply:
I was implying that should be in the docs.

Reply:
Updated.